### PR TITLE
Improve sync docs in the testsuite

### PR DESCRIPTION
**Pull request by psychon**
Here is my attempt at fixing #4365. Please double check what I did here, because:

- I am describing something that I learnt in the last few days by reading source code (no tests were run!). Thus, it is possible that my understanding is wrong.
- I had to use `dia` to change images. I have no idea what I am doing here (but felt that it is necessary to add the IPC response to the images). (Please don't make me touch `dia` again...)
- I have no idea what format is used for these docs and was too lazy to figure this out. Thanks to this I noticed that I am way too used to markdown these days.

### CODE REVIEW

File: docs/testsuite [hunk 1]
```diff
@@ -569,7 +569,35 @@ i3 will then reply with the same ClientMessage, sent to the window specified in
...
+This protocol is used, for example, in t/525-i3bar-mouse-bindings.t: A mouse
+button press on i3bar is triggered. i3bar reacts to this by sending IPC commands
+to i3.
+
+The necessary synchronisation is achieved by sending an I3_SYNC event to i3bar:
+Because i3bar reacts with a sync IPC command to i3, all previous IPC commands from
+i3bar will be handled first. Because i3 reacts via X11, all previous X11
+requests from i3 will be handled by the X11 server first.
+
+The actual test also has to sync with i3 first due to how X11 handling works. If
```
Comment (Airblader) [inline docs/testsuite]: Can we please phrase this in a less dismissive way? :-)

> For more details refer to the documentation for XAllowEvents with mode ReplayPointer

**Review by Airblader — COMMENTED** [review]
The images were previously *.png, now *.dia – how come? GH can't open them, so I haven't yet looked at them.

**Comment by psychon** [comment]
> The images were previously *.png, now *.dia – how come? 

The images were always `*.dia` plus a `*.png` exported from `dia`. But GH is clearly now showing this properly. Its claim of "empty file" does not help...

This is how `git show` shows the situation:
```
diff --git a/docs/i3-sync-working.dia b/docs/i3-sync-working.dia
index 9f1c3bc8..25803828 100644
Binary files a/docs/i3-sync-working.dia and b/docs/i3-sync-working.dia differ
diff --git a/docs/i3-sync-working.png b/docs/i3-sync-working.png
index dce44ac5..419334d0 100644
Binary files a/docs/i3-sync-working.png and b/docs/i3-sync-working.png differ
diff --git a/docs/i3-sync.dia b/docs/i3-sync.dia
index 0945ae26..c8fb8359 100644
Binary files a/docs/i3-sync.dia and b/docs/i3-sync.dia differ
diff --git a/docs/i3-sync.png b/docs/i3-sync.png
index b64cce25..c9a5a01c 100644
Binary files a/docs/i3-sync.png and b/docs/i3-sync.png differ
```

File: docs/testsuite (hunk 1 above)
Comment (psychon) [reply docs/testsuite]: Thanks for the suggestion. Done. (In a new commit - dunno if you want me to amend the existing commit instead.)

**Review by Airblader — COMMENTED** [review]
I actually just read GH wrong – red color for indicating reduced file size. OK then.

Wondering if @stapelberg wants to give this a read? It all sounds good to me but frankly I'm not very deeply invested in this sync protocol.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Looks good, thanks for this!

> * I have no idea what format is used for these docs and was too lazy to figure this out. Thanks to this I noticed that I am way too used to markdown these days.

It’s asciidoc, but we’re switching to markdown in https://github.com/i3/i3/issues/4327 :)

**Pull request merged**